### ISSUE TITLE
Use BackgroundMode.transparent make Crash

### ISSUE THREAD

**Issue by yuhaocan** [flutter/flutter#57096]
Update FlutterSDK to 1.17.0 and use io.flutter.embedding.android.FlutterActivity.Then I use BackgroundMode.transparent make crash.
* log
```
2020-05-13 16:25:16.390 26523-26523/com.example.flutter_demo E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.example.flutter_demo, PID: 26523
    java.lang.RuntimeException: Unable to start activity ComponentInfo{com.example.flutter_demo/com.example.flutter_demo.MainActivity}: java.lang.NullPointerException: Attempt to invoke interface method 'void io.flutter.embedding.engine.renderer.RenderSurface.attachToRenderer(io.flutter.embedding.engine.renderer.FlutterRenderer)' on a null object reference
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2946)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3081)
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:78)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:108)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:68)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1831)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:201)
        at android.app.ActivityThread.main(ActivityThread.java:6806)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:547)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:873)
     Caused by: java.lang.NullPointerException: Attempt to invoke interface method 'void io.flutter.embedding.engine.renderer.RenderSurface.attachToRenderer(io.flutter.embedding.engine.renderer.FlutterRenderer)' on a null object reference
        at io.flutter.embedding.android.FlutterView.attachToFlutterEngine(FlutterView.java:699)
        at io.flutter.embedding.android.FlutterActivityAndFragmentDelegate.onCreateView(FlutterActivityAndFragmentDelegate.java:290)
        at io.flutter.embedding.android.FlutterActivity.createFlutterView(FlutterActivity.java:519)
        at io.flutter.embedding.android.FlutterActivity.onCreate(FlutterActivity.java:413)
        at com.example.flutter_demo.MainActivity.onCreate(MainActivity.java:14)
        at android.app.Activity.performCreate(Activity.java:7224)
        at android.app.Activity.performCreate(Activity.java:7213)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1272)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2926)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3081) 
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:78) 
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:108) 
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:68) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1831) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loop(Looper.java:201) 
        at android.app.ActivityThread.main(ActivityThread.java:6806) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:547) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:873) 

```

**Expected results:** 
launcher App open FlutterActivity will crash.
**Actual results:** 
launcher App open FlutterActivity success.
<details>
  <summary>Logs</summary>

```
```

```

```

```
[✓] Flutter (Channel unknown, v1.17.0, on Mac OS X 10.15.4 19E287, locale zh-Hans-CN)
    • Flutter version 1.17.0 at /Users/julyyu1/Library/Flutter/1.17.0/flutter
    • Framework revision e6b34c2b5c (11 days ago), 2020-05-02 11:39:18 -0700
    • Engine revision 540786dd51
    • Dart version 2.8.1

 
[!] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/julyyu1/SDK/android-sdk-macosx
    • Platform android-29, build-tools 29.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)
    ! Some Android licenses not accepted.  To resolve this, run: flutter doctor --android-licenses

[✓] Xcode - develop for iOS and macOS (Xcode 11.4.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.4.1, Build version 11E503a
    • CocoaPods version 1.9.1

[✓] Android Studio (version 3.5)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 44.0.1
    • Dart plugin version 191.8593
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)

[!] VS Code (version 1.45.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    ✗ Flutter extension not installed; install from
      https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected device (1 available)
    • MIX 2S • a6d1ab7b • android-arm64 • Android 9 (API 28)

! Doctor found issues in 2 categories.

```

</details>

**Comment by VladyslavBondarenko** [comment]
Hi @yuhaocan 
Please see https://github.com/flutter/flutter/issues/56125,
Closing since the issue is fixed in `1.18.0-dev.0.0` (but the fix is not yet in stable)

**Comment by lslv1243** [comment]
There is a workaround to make it work on Flutter 1.17. Subclass FlutterActivity and override the method `provideSplashScreen`. It is a very ugly hack, but it does work.

```
override fun provideSplashScreen(): SplashScreen? {
    // MARK: Hack to fix problem with transparent Flutter Activity
    // it is fixed on Flutter 1.18, but it still does not have
    // a stable version available
    val flutterViewField = delegate.javaClass.getDeclaredField("flutterView")
    flutterViewField.isAccessible = true
    val flutterView = flutterViewField.get(delegate) as io.flutter.embedding.android.FlutterView
    val renderSurfaceField = flutterView.javaClass.getDeclaredField("renderSurface")
    renderSurfaceField.isAccessible = true
    val flutterTextureViewField = flutterView.javaClass.getDeclaredField("flutterTextureView")
    flutterTextureViewField.isAccessible = true
    renderSurfaceField.set(flutterView, flutterTextureViewField.get(flutterView))

    return super.provideSplashScreen()
}
```

How it works: This method is called after the flutterView is created and before it is attached to the engine. So I overrided this method to fix the wrong property.

**Comment by FLYINGKRIPTO** [comment]
> There is a workaround to make it work on Flutter 1.17. Subclass FlutterActivity and override the method `provideSplashScreen`. It is a very ugly hack, but it does work.
> 
> ```
> override fun provideSplashScreen(): SplashScreen? {
>     // MARK: Hack to fix problem with transparent Flutter Activity
>     // it is fixed on Flutter 1.18, but it still does not have
>     // a stable version available
>     val flutterViewField = delegate.javaClass.getDeclaredField("flutterView")
>     flutterViewField.isAccessible = true
>     val flutterView = flutterViewField.get(delegate) as io.flutter.embedding.android.FlutterView
>     val renderSurfaceField = flutterView.javaClass.getDeclaredField("renderSurface")
>     renderSurfaceField.isAccessible = true
>     val flutterTextureViewField = flutterView.javaClass.getDeclaredField("flutterTextureView")
>     flutterTextureViewField.isAccessible = true
>     renderSurfaceField.set(flutterView, flutterTextureViewField.get(flutterView))
> 
>     return super.provideSplashScreen()
> }
> ```
> 
> How it works: This method is called after the flutterView is created and before it is attached to the engine. So I overrided this method to fix the wrong property.

Doesn't work for me!! on 1.17.5

**Comment by lslv1243** [comment]
> > There is a workaround to make it work on Flutter 1.17. Subclass FlutterActivity and override the method `provideSplashScreen`. It is a very ugly hack, but it does work.
> > ```
> > override fun provideSplashScreen(): SplashScreen? {
> >     // MARK: Hack to fix problem with transparent Flutter Activity
> >     // it is fixed on Flutter 1.18, but it still does not have
> >     // a stable version available
> >     val flutterViewField = delegate.javaClass.getDeclaredField("flutterView")
> >     flutterViewField.isAccessible = true
> >     val flutterView = flutterViewField.get(delegate) as io.flutter.embedding.android.FlutterView
> >     val renderSurfaceField = flutterView.javaClass.getDeclaredField("renderSurface")
> >     renderSurfaceField.isAccessible = true
> >     val flutterTextureViewField = flutterView.javaClass.getDeclaredField("flutterTextureView")
> >     flutterTextureViewField.isAccessible = true
> >     renderSurfaceField.set(flutterView, flutterTextureViewField.get(flutterView))
> > 
> >     return super.provideSplashScreen()
> > }
> > ```
> > 
> > 
> > How it works: This method is called after the flutterView is created and before it is attached to the engine. So I overrided this method to fix the wrong property.
> 
> Doesn't work for me!! on 1.17.5

did you put extra to your intent? 
```
it.putExtra("background_mode"/*FlutterActivityLaunchConfigs.EXTRA_BACKGROUND_MODE*/, FlutterActivityLaunchConfigs.BackgroundMode.transparent.name)
```

**Comment by xuzhongpeng** [comment]
> 

work for me in 1.17.5-candidate.3
here is my code
```
override fun provideSplashScreen(): SplashScreen? {
        try {
            val clazz: Class<*> = this.javaClass
            val delegateField = clazz.superclass!!.getDeclaredField("delegate")
            delegateField.isAccessible = true
            val delegate = delegateField.get(this)
            val flutterViewField = delegate.javaClass.getDeclaredField("flutterView")
            flutterViewField.isAccessible = true
            val flutterView = flutterViewField.get(delegate)
            val renderSurfaceField = flutterView.javaClass.getDeclaredField("renderSurface")
            renderSurfaceField.isAccessible = true
            val flutterTextureViewField = flutterView.javaClass.getDeclaredField("flutterTextureView")
            flutterTextureViewField.isAccessible = true
            renderSurfaceField.set(flutterView, flutterTextureViewField.get(flutterView))
        } catch (e: IllegalArgumentException) {

        }
        return super.provideSplashScreen()
    }
```
FlutterEngine init
```
CachedEngineFragmentBuilder builder = withCachedEngine(REPORT_FLUTTER_ENGINE);
            try {
               // here is make the fragmentClass to my own class(GmFlutterFragment.class)
                Class clazz = Class.forName("io.flutter.embedding.android.FlutterFragment$CachedEngineFragmentBuilder");
                Constructor<?> constructorParameter = clazz.getDeclaredConstructor(Class.class,String.class);
                Object obj = constructorParameter.newInstance(GmFlutterFragment.class, REPORT_FLUTTER_ENGINE);
                builder = (CachedEngineFragmentBuilder) obj;
            } catch (IllegalAccessException | ClassNotFoundException | InvocationTargetException | java.lang.InstantiationException | NoSuchMethodException e) {
                e.printStackTrace();
            }
            fragment = builder
                    .renderMode(RenderMode.texture)
                    .build();
```
